main/states/base_state.py

```diff
@@ -11,7 +11,8 @@ class State(ABC):
     """State is an abstract class to represent a state. It defines the abstract method for behavior of state and
     transition method to facilitate transition between states.
     """
-
+    video_process = None
```

Comment:
Follow a getter/setter approach here and use the `@property` i.e. declare these in init and do not access these directly by the class name.